cdap-security/src/main/java/co/cask/cdap/security/authorization/RemoteAuthorizationEnforcer.java

```diff
@@ -102,8 +106,8 @@ public void enforce(EntityId entity, Principal principal, Action action) throws
 
   @Override
   public Set<? extends EntityId> isVisible(Set<? extends EntityId> entityIds, Principal principal) throws Exception {
-    // TODO: needs to be implemented
-    throw new UnsupportedOperationException("This method needs to be implemented!");
+    // TODO: figure out how to cache the result
```

Comment:
We can have another cache called `visibilityCache` that has a mapping between `EntitiyId` => `Boolean`